### PR TITLE
Implement better upload with upload sessions

### PR DESCRIPTION
**Pull request by mirkomilova10**
Upload session for uploading files larger than 4mib
It works on my end but theres some error handling than needs to be done/fixed i think
Also changed the API endpoint to use Microsoft's Graph

### CODE REVIEW

**Comment by dkatavic** [comment]
@mirkomilova10 thanks for the PR. As we are moving from onedrive API to MS Graph, this could potentially have breaking changes, so I would need to further check out will everything still work. I have run the tests and most of the APIs are working, I just need to figure out why I am getting random timeouts

Did you test the APIs with graph and is everything working for you?

ps. can you increase the version in `package.json` to `0.1.0`

**Comment by mirkomilova10** [comment]
I think that the API is mostly the same, the only difference is the endpoint which is graph now. Everything else should be working the same as it did before
Also the random timeouts may be due to the accessToken expiring very fast (less than an hour)

File: lib/items/uploadSession.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+              callback();
+          }, st.stime);
+      });
+    });
+  });
+}
+
+function getparams(filePath){
+    var allsize = fs.statSync(filePath).size;
+    var sep = allsize < (60 * 1024 * 1024) ? allsize : (60 * 1024 * 1024) - 1;
```
Comment (dkatavic) [inline lib/items/uploadSession.js]: If I got it right, you are checking is file size < 60MB? can you add a comment here to explain what is happening?

File: lib/items/uploadSession.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    var ar = [];
+    for (var i = 0; i < allsize; i += sep) {
+        var bstart = i;
+        var bend = i + sep - 1 < allsize ? i + sep - 1 : allsize - 1;
+        var cr = 'bytes ' + bstart + '-' + bend + '/' + allsize;
+        var clen = bend != allsize - 1 ? sep : allsize - i;
+        var stime = allsize < (60 * 1024 * 1024) ? 5000 : 10000;
```
Comment (dkatavic) [inline lib/items/uploadSession.js]: can you explain what is `stime`?

**Comment by dkatavic** [comment]
Closing because of https://github.com/dkatavic/onedrive-api/pull/15

**Pull request closed without merging**